core/src/main/scala/akka/kafka/scaladsl/Producer.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Sink
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import akka.kafka.internal.ProducerStage
+import akka.stream.scaladsl.Keep
+import akka.kafka.ProducerSettings
+
+object Producer {
+
+  final case class Message[K, V, +PassThrough](
+    record: ProducerRecord[K, V],
+    passThrough: PassThrough
+  )
+
+  final case class Result[K, V, PassThrough](
+    offset: Long,
+    message: Message[K, V, PassThrough]
+  )
+
+  /**
+   * Publish records to Kafka topics.
+   */
+  def plainSink[K, V](settings: ProducerSettings[K, V]): Sink[ProducerRecord[K, V], NotUsed] =
+    Flow[ProducerRecord[K, V]].map(record => Message(record, NotUsed))
+      .viaMat(flow(settings))(Keep.right)
+      .to(Sink.ignore)
+
+  /**
+   * Sink that is aware of the committable offset from a [[Consumer]]. It will commit the consumer offset
+   * when the message has been published successfully to the topic.
+   */
+  def commitableSink[K, V](settings: ProducerSettings[K, V]): Sink[Message[K, V, Consumer.Committable], NotUsed] =
+    flow[K, V, Consumer.Committable](settings)
+      .mapAsync(1)(identity)
+      .mapAsync(1)(_.message.passThrough.commit())
+      .to(Sink.ignore)
+
+  /**
+   * Publish records to Kafka topics and then continue the flow. Possibility to pass through a message, which
+   * can for example be a [[Consumer.CommittableOffset]] or [[Consumer.CommittableOffsetBatch]] that can
+   * be committed later in the flow.
+   */
+  def flow[K, V, PassThrough](settings: ProducerSettings[K, V]): Flow[Message[K, V, PassThrough], Future[Result[K, V, PassThrough]], NotUsed] =
+    Flow.fromGraph(new ProducerStage[K, V, PassThrough](() => settings.creteKafkaProducer())) //FIXME is that Future needed?
```

Comment:
AFAIU the point is that the thing that people will do is `mapAsync(1): 15804 ms` which is 10x slower than all the other ones, so having the future there exposes a risk of people panicking about "omg it's so slow" – if I understood correctly